### PR TITLE
Clean up event listeners on anim clip destroy

### PR DESCRIPTION
**Pull request by yaustar**
Fixes https://github.com/playcanvas/engine/issues/5259

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/components/sprite/sprite-animation-clip.js [hunk 1]
```diff
@@ -289,7 +289,7 @@ class SpriteAnimationClip extends EventHandler {
         asset.off('remove', this._onSpriteAssetRemove, this);
 
         // unbind atlas
-        if (asset.resource && asset.resource.atlas) {
+        if (asset.resource && !asset.resource.atlas) {
```
Comment (yaustar) [inline src/framework/components/sprite/sprite-animation-clip.js:292]: We only set the event listener if `.atlas` is null (see https://github.com/playcanvas/engine/blob/main/src/framework/components/sprite/sprite-animation-clip.js#L303)

So it makes sense to only remove it if it is still null (ie the texture atlas hasn't been loaded yet)

**Review by willeastcott — APPROVED** [review]

**Comment by willeastcott** [comment]
Would be good to fully verify this fix before merging.... (nice fix tho!)

**Comment by yaustar** [comment]
I've asked the client to send over details where we can test this fix in a larger project with a large number of sprites, scene changing etc.

**Comment by yaustar** [comment]
Checked against the clients larger project and no visual issues can be seen in areas where there's heavy sprite usuage

**Pull request merged**